services/build_service.rb

```diff
@@ -15,11 +15,39 @@ def initialize(build_data_source: nil)
     end
 
     def list_builds(project: nil)
-      self.build_data_source.list_builds(project: project)
+      return self.build_data_source.list_builds(project: project)
     end
 
     def pending_builds(project: nil)
-      self.build_data_source.pending_builds(project: project)
+      return self.build_data_source.pending_builds(project: project)
+    end
+
+    # returns a list of commit shas where the status is pending
+    # and it hasn't been superceeded by a newer build
+    def pending_build_shas_needing_rebuilds(project:)
```

Comment:
@minuscorp is this functional enough? I can't tell yet 😆

Reply:
**MORE** function chaining